artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -461,7 +463,8 @@ private void doConnect() {
          senders.clear();
          receivers.clear();
 
-         ClientSASLFactory saslFactory = new SaslFactory(connection, brokerConnectConfiguration);
+         final String[] availableSaslMechanisms = configuration.getExtraParams().containsKey(SASL_MECHANISMS_KEY) ? protonProtocolManager.getSaslMechanisms() : null;
```

Comment:
The array `availableSaslMechanisms` seems more the list of the enabled mechanisms than the available mechanisms:
```suggestion
         final String[] enabledSaslMechanisms = configuration.getExtraParams().containsKey(SASL_MECHANISMS_KEY) ? protonProtocolManager.getSaslMechanisms() : null;
```